llvm/lib/Transforms/Vectorize/VPlanCFG.h

```diff
@@ -306,6 +306,15 @@ template <> struct GraphTraits<VPlan *> {
   }
 };
 
+inline auto VPlan::getExitBlocks() {
+  VPBlockBase *ScalarHeader = getScalarHeader();
+  return make_filter_range(
+      VPBlockUtils::blocksOnly<VPIRBasicBlock>(
+          vp_depth_first_shallow(getVectorLoopRegion()->getSingleSuccessor())),
+      [ScalarHeader](VPIRBasicBlock *VPIRBB) {
+        return VPIRBB != ScalarHeader && VPIRBB->getNumSuccessors() == 0;
```

Comment:
Same comment as in the other PR - I think this can return different blocks depending upon whether you call this before or during VPlan::execute. Is this something we should worry about?

Reply:
Ah, I think the reason why this works is because of the extra `VPIRBB->getNumSuccessors() == 0` condition, since the middle block will always have at least one successor.

Reply:
The middle-block may or may not be included by `VPBlockUtils::blocksOnly<VPIRBasicBlock>`, but the middle block will never be considered as an exit block, as it has successors.

Reply:
Yeah, thanks for explaining! I realised that around the same time that you replied. It just took me a while to figure it out. :)